### PR TITLE
darkside: Store, track, and expose commitment tree sizes

### PR DESCRIPTION
**Pull request by str4d**
Please ensure this checklist is followed for any pull requests for this repo. This checklist must be checked by both the PR creator and by anyone who reviews the PR.
* [ ] Relevant documentation for this PR has to be completed before the PR can be merged
* [ ] A test plan for the PR must be documented in the PR notes and included in the test plan for the next regular release

As a note, all CI tests need to be passing and all appropriate code reviews need to be done before this PR can be merged

### CODE REVIEW

File: common/darkside.go [hunk 1]
```diff
@@ -248,7 +294,12 @@ func DarksideApplyStaged(height int) error {
 		}
 		block[68]++ // hack HashFinalSaplingRoot to mod the block hash
 		block = append(block, tx.bytes...)
-		state.activeBlocks[tx.height-state.startHeight] = block
+		state.activeBlocks[tx.height-state.startHeight].bytes = block
+		// Now increment this and every subsequent block's commitment tree sizes.
+		for _, b := range state.activeBlocks[tx.height-state.startHeight:] {
+			b.saplingTreeSize += uint32(tx.saplingOutputs)
+			b.orchardTreeSize += uint32(tx.orchardActions)
+		}
```
Comment (str4d) [inline common/darkside.go:302]: In local testing I'm getting some off-by-twos, and while I haven't identified the source yet, I expect I have made a mistake in this change.

**Comment by LarryRuane** [comment]
Your first commit regenerates the generated files with the latest tools -- I'm just wondering your opinion, should I remove these files from source control (and document how to install the tools to generate them)? Seems to me that would be better. But I guess the risk is reviewers, testers, users may have different versions of the tools and thus get different results.

**Review by LarryRuane — COMMENTED** [review]
Looks good, aside from the one suggestion.

**Comment by str4d** [comment]
> Your first commit regenerates the generated files with the latest tools -- I'm just wondering your opinion, should I remove these files from source control (and document how to install the tools to generate them)? Seems to me that would be better. But I guess the risk is reviewers, testers, users may have different versions of the tools and thus get different results.

I was forced to use newer tools because I fetch my tools from Homebrew, and there is no way to install specific versions. I think pinning the generated files still likely makes sense; we shouldn't be changing the protobuf files as frequently as the rest of the code, so we can just require that anyone changing them is using the newest version of the generation tools.

**Comment by LarryRuane** [comment]
ACK c0db49b53e2fefe0d9e88674d5543c1fa1330a56

**Pull request merged**